hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/HoodieInputFormat.java

```diff
@@ -218,4 +215,109 @@ protected static HoodieTableMetaClient getTableMetaClient(FileSystem fs, Path da
     LOG.info("Reading hoodie metadata from path " + baseDir.toString());
     return new HoodieTableMetaClient(fs.getConf(), baseDir.toString());
   }
+
+  /**
+   * Achieves listStatus functionality for an incrementally queried table. Instead of listing all
+   * partitions and then filtering based on the commits of interest, this logic first extracts the
+   * partitions touched by the desired commits and then lists only those partitions.
+   */
+  private List<FileStatus> listStatusForIncrementalMode(JobConf job,
+      HoodieTableMetaClient tableMetaClient, List<Path> inputPaths) throws IOException {
+    String tableName = tableMetaClient.getTableConfig().getTableName();
+    HoodieTimeline timeline = tableMetaClient.getActiveTimeline().getCommitsTimeline()
```

Comment:
HoodieInputFormat supports only RO based tables, which means we should use tableMetaClient.getActiveTimeline().getCommitTimeline() instead of tableMetaClient.getActiveTimeline().getCommitsTimeline(). But that brings another question, what is the plan for HoodieRealtimeInputFormat ? Are we planning to do similar optimizations there ?

Reply:
This is already achieved, since HoodieRealtimeInputFormat extends HoodieInputFormat and HoodieRealtimeInputFormat.listStatus just does super.listStatus.